examples/workflow-test/src/graph/workflow.graph.ts

```diff
@@ -0,0 +1,16 @@
+import { GLSPSemanticGraph, PNode, PNodeConstructor } from '@eclipse-glsp/glsp-playwright-core/glsp';
+import { waitForFunction } from '@eclipse-glsp/glsp-playwright-core/integration';
+
+export class WorkflowGraph extends GLSPSemanticGraph {
+    override async waitForCreationOfNodeType<TElement extends PNode>(
+        constructor: PNodeConstructor<TElement>,
+        creator: () => Promise<void>
+    ): Promise<TElement[]> {
+        const elements = await super.waitForCreationOfNodeType(constructor, creator);
+
+        await waitForFunction(async () =>
+            elements.every(element => element.locate().evaluate(node => node?.classList.contains('selected')))
```

Comment:
Why do we need to wait for all nodes to be selected? Or maybe I'm totally mis-reading this.

Reply:
You are not mis-reading it. After creating a node it gets the `selected` class after "some" time. This time discrepancy between creating the element and being visible (i.e., not ready) and being ready (i.e., fully rendered) makes it necessary to check for it.